packages/nimble-components/src/table/models/keyboard-navigation-manager.ts

```diff
@@ -0,0 +1,1289 @@
+                    focusType: TableFocusType.cellContent,
+                    columnIndex: cellIndex,
+                    cellContentIndex: i
+                });
+                if (
+                    this.focusType === TableFocusType.cellContent
+                    && this.columnIndex === cellIndex
+                    && this.cellContentIndex === i
+                ) {
+                    startIndex = focusStates.length - 1;
+                }
+            }
+            if (cellInfo.actionMenuButton) {
+                focusStates.push({
+                    focusType: TableFocusType.cellActionMenu,
+                    columnIndex: cellIndex
+                });
+                if (
+                    this.focusType === TableFocusType.cellActionMenu
+                    && this.columnIndex === cellIndex
+                ) {
+                    startIndex = focusStates.length - 1;
+                }
+            }
+            const lastCellTabbableIndex = focusStates.length - 1;
+            if (
+                this.focusType === TableFocusType.cell
+                && this.columnIndex === cellIndex
+            ) {
+                startIndex = shiftKeyPressed
+                    ? lastCellTabbableIndex + 1
+                    : firstCellTabbableIndex - 1;
+            }
+            cellIndex += 1;
+        }
+        if (this.focusType === TableFocusType.row) {
+            startIndex = shiftKeyPressed ? focusStates.length : -1;
+        }
+        const direction = shiftKeyPressed ? -1 : 1;
+        return focusStates[startIndex + direction];
+    }
+
+    private getNextHeaderTabStop(
+        shiftKeyPressed: boolean
+    ): TableFocusState | undefined {
+        let startIndex = -1;
+        const focusStates = [];
+        const headerTabbableElements = this.getTableHeaderFocusableElements().headerActions;
+        for (let i = 0; i < headerTabbableElements.length; i++) {
+            focusStates.push({
+                focusType: TableFocusType.headerActions,
+                headerActionIndex: i
+            });
+        }
+        if (this.focusType === TableFocusType.headerActions) {
+            startIndex = this.headerActionIndex;
+        } else {
+            // TableFocusType.columnHeader
+            startIndex = focusStates.length;
+        }
+        const direction = shiftKeyPressed ? -1 : 1;
+        return focusStates[startIndex + direction];
+    }
+
+    private blurAfterLastTab(activeElement: HTMLElement): void {
+        // In order to get the desired browser-provided Tab/Shift-Tab behavior of focusing the
+        // element before/after the table, the table shouldn't have tabIndex=0 when this event
+        // handling ends. However it needs to be tabIndex=0 so we can re-focus the table the next time
+        // it's tabbed to, so set tabIndex back to 0 after a rAF.
+        // Note: In Chrome this is only needed for Shift-Tab, but in Firefox both Tab+Shift-Tab need this
```

Comment:
nit: reads kind of like a funky multi-step keyboard combination if we don't spell it out
```suggestion
        // Note: In Chrome this is only needed for Shift-Tab, but in Firefox both Tab and Shift-Tab need this
```

Reply:
Updated